### ISSUE TITLE
Markdown editor needs work

### ISSUE THREAD

**Issue by tremby** [GraphCMS/Issue-tracker#35]
The markdown editor seems dangerous to me.

In no more than three minutes of playing with it I made it do some unexpected things.

- Red bullet points turn up sometimes in the code view but don't seem to correspond to anything in the rendered view.
- A newline entered in the rendered view (with shift-enter) doesn't seem to have any corresponding output in the code view (could be a line terminated with two spaces, or a literal `<br>`, for example). Going back and forth and making edits eventually kills the newline.
- Meanwhile a newline in the code view causes no line break in the rendered view (this is correct, at least for [Commonmark](http://commonmark.org/)), but again flipping back and forth between views and making edits this markdown newline sometimes gets removed.
- I'm getting extra empty paragraphs appearing out of nowhere when flipping back and forth between views.

To put it simply, what I write in the code view in Markdown should not be changed and rewritten by the rendered view.

In fact, the whole point of Markdown is to write plain text and get out styled text. In my opinion there should be no editable rendered view at all. A Markdown editor should be a Markdown editor, not a wysiwyg editor which outputs Markdown. At the very least there should be an option to disallow editing in the rendered view.

**Comment by brandonmp** [comment]
+1.  please fix this, markdown editor is unusable in current state. 

To add specifics to OP: tables are also pretty broken. Just editing / saving a post with a table seems act of saving a blog post with edits destroys table, apparently removing all the line returns & quietly breaking the post.

Text before making edits not related to the table:
```
| School     | GPA     | GMAT     |  
| ---------- | ---------- | ---------- |  
| Haas       | 3.7    | 770       |  
| Kellogg       | 3.2     | 720    |  
| Wharton       | 3.2    | 710     |  
| Duke       | 3.1   | 680     |  
| UVA       | 3.5       | 720  |
```
Text after saving unrelated edits:
```
| School     | GPA     | GMAT     |  | ---------- | ---------- | ---------- |  | Haas       | 3.7    | 770       |  | Kellogg       | 3.2     | 720    |  | Wharton       | 3.2    | 710     |  | Duke       | 3.1   | 680     |  | UVA       | 3.5       | 720  |
```

which, obviously my markdown parser doesn't render into an html table.

**Comment by brandonmp** [comment]
Wow, okay, one more problem to add: links with relative paths are saved as relative to `app.graphcms.com` with--again, absolutely no indication that my inputted value is being overridden.

i.e.,
`[anchortext](/some-path/)`

ends up in my app as 
`[anchortext](/https://app.graphcms.com/some-path/)`

..and trying to fix that broke my tables again.

**Comment by mlukaszczyk** [comment]
Hey all! We are soon going to switch rich and markdown editors completely. Sorry for the inconvenience!
